code/datums/components/rust.dm

```diff
@@ -0,0 +1,87 @@
+/**
+ * Adding this component to an Atom will have it automatically render an overlay.
+ * The overlay can be specified in new as the first and only parameter; it defaults to "rust" if not included.
+ */
+/datum/component/rust
+	dupe_mode = COMPONENT_DUPE_UNIQUE
+	/// Internal variable to store the rust overlay we are using to avoid regenerating it every overlay update call
+	var/mutable_appearance/rust_overlay
+
+/turf/closed/wall/rust/New()
+	var/atom/wall_new = new /turf/closed/wall(src)
+	wall_new._AddComponent(list(/datum/component/rust))
+
+/turf/closed/wall/r_wall/rust/New()
+	var/atom/wall_new = new /turf/closed/wall/r_wall(src)
+	wall_new._AddComponent(list(/datum/component/rust))
+
+/turf/open/floor/plating/rust/New()
+	var/atom/wall_new = new /turf/open/floor/plating(src)
+	wall_new._AddComponent(list(/datum/component/rust))
+
+/datum/component/rust/Initialize(rust_iconstate = "rust")
+	. = ..()
+	if(!isatom(parent))
+		stack_trace("Attempted to intialize a Rust Component for non-Atom [parent]")
+		return COMPONENT_INCOMPATIBLE
+	var/atom/parent_atom = parent
+	if(!(rust_iconstate in icon_states(parent_atom.icon)))
+		stack_trace("Attempted to intialize a Rust Component for [parent] with invalid rust iconstate '[rust_iconstate]'")
+		return COMPONENT_INCOMPATIBLE
+
+	rust_overlay = mutable_appearance(parent_atom.icon, rust_iconstate)
+
+/datum/component/rust/RegisterWithParent()
+	RegisterSignal(parent, COMSIG_ATOM_UPDATE_OVERLAYS, .proc/apply_rust_overlay)
+	RegisterSignal(parent, list(COMSIG_ATOM_SECONDARY_TOOL_ACT(TOOL_WELDER), COMSIG_ATOM_SECONDARY_TOOL_ACT(TOOL_RUSTSCRAPER)), .proc/secondary_tool_act)
+	RegisterSignal(parent, COMSIG_PARENT_PREQDELETED, .proc/parent_del)
+	RegisterSignal(parent, COMSIG_PARENT_EXAMINE, .proc/handle_examine)
+
+/datum/component/rust/UnregisterFromParent()
+	UnregisterSignal(parent,\
+		list(COMSIG_ATOM_UPDATE_OVERLAYS, COMSIG_ATOM_SECONDARY_TOOL_ACT(TOOL_WELDER), COMSIG_ATOM_SECONDARY_TOOL_ACT(TOOL_RUSTSCRAPER)), COMSIG_PARENT_PREQDELETED)
+
+/datum/component/rust/proc/handle_examine(datum/source, mob/user, list/examine_text)
+	SIGNAL_HANDLER
+	examine_text += span_notice("The [source] is very rusty; you could probably <i>burn</i> or <i>scrape</i> it off.")
```

Comment:
```suggestion
	examine_text += span_notice("[source] is very rusty, you could probably <i>burn</i> or <i>scrape</i> it off.")
```

"The" is implicit in BYOND. The semicolon thing is just for consistency across other messages, which I don't think have them.